### PR TITLE
ebrelayer fixes to move eth => ceth for 0.42

### PR DESCRIPTION
**Pull request by banshee**
This branch has some changes that clearly need to come out (logging, etc) but I thought I'd go ahead and start with a PR for it.

### CODE REVIEW

File: app/app.go [hunk 1]
```diff
@@ -130,6 +131,7 @@ var (
 
 		clptypes.ModuleName:     {authtypes.Burner, authtypes.Minter},
 		dispensation.ModuleName: {authtypes.Burner, authtypes.Minter},
+		ethbridge.ModuleName:    {authtypes.Burner, authtypes.Minter},
```
Comment (banshee) [inline app/app.go:134]: Is this the right thing to do?  It definitely works, but it seems strange it wasn't here already

File: app/app.go [hunk 2]
```diff
@@ -467,6 +469,8 @@ func (app *SifchainApp) InitChainer(ctx sdk.Context, req abci.RequestInitChain)
 		panic(err)
 	}
 
+	genesisState["ethbridge"] = json.RawMessage{}
```
Comment (banshee) [inline app/app.go:472]: This is strange.  It's just a hack to get InitGenesis called - if the genesis data for "ethbridge" is empty, InitGenesis isn't called at all, and that's where we set up:

```
	bridgeAccount := authtypes.NewEmptyModuleAccount(ModuleName, authtypes.Burner, authtypes.Minter)
	am.AccountKeeper.SetModuleAccount(ctx, bridgeAccount)
```

Comment (juniuszhou) [reply app/app.go:472]: don't understand, why just clean the the genesis state of this module, if also need clean other module like oracle, clp.

File: app/app.go (hunk 1 above)
Comment (timlind) [reply app/app.go:134]: Must've missed this one, ethbridge was one of the last modules we re-added once it was migrated.

File: x/ethbridge/client/cli/tx.go [hunk 3]
```diff
@@ -30,7 +30,12 @@ func GetCmdCreateEthBridgeClaim() *cobra.Command {
 
 			flags := cmd.Flags()
 
-			ethereumChainID, err := flags.GetInt64(types.FlagEthereumChainID)
+			ethereumChainIDStr, err := flags.GetString(types.FlagEthereumChainID)
```
Comment (banshee) [inline x/ethbridge/client/cli/tx.go:33]: Why can't we just use GetInt64 here?

File: cmd/ebrelayer/main.go [hunk 4]
```diff
@@ -206,8 +212,14 @@ func RunInitRelayerCmd(cmd *cobra.Command, args []string) error {
 	zap.RedirectStdLog(sugaredLogger.Desugar())
 
 	// Initialize new Ethereum event listener
+	//validatorAddress, err := relayer.AddToKeyringWithMnemonic(cliContext.Keyring, validatorMoniker, mnemonic)
+	//if err != nil {
+	//	panic("could not add relayer to keyring")
+	//}
+	//
+	//valAddrAsFrom := sdk.ValAddress(validatorAddress.GetAddress())
```
Comment (ElliotFriedman) [inline cmd/ebrelayer/main.go:220]: Let's remove commented code

File: cmd/ebrelayer/main.go [hunk 5]
```diff
@@ -187,7 +193,7 @@ func RunInitRelayerCmd(cmd *cobra.Command, args []string) error {
 		return errors.Errorf("invalid [validator-moniker]: %s", args[3])
 	}
 	validatorMoniker := args[3]
-	mnemonic := args[4]
+	//mnemonic := args[4]
```
Comment (ElliotFriedman) [inline cmd/ebrelayer/main.go:196]: Let's remove commented code

File: cmd/ebrelayer/main.go [hunk 6]
```diff
@@ -206,8 +212,14 @@ func RunInitRelayerCmd(cmd *cobra.Command, args []string) error {
...
 	// Initialize new Ethereum event listener
+	//validatorAddress, err := relayer.AddToKeyringWithMnemonic(cliContext.Keyring, validatorMoniker, mnemonic)
+	//if err != nil {
+	//	panic("could not add relayer to keyring")
+	//}
+	//
+	//valAddrAsFrom := sdk.ValAddress(validatorAddress.GetAddress())
 	ethSub, err := relayer.NewEthereumSub(cliContext, rpcURL, validatorMoniker, chainID, web3Provider,
-		contractAddress, privateKey, mnemonic, db, sugaredLogger)
+		contractAddress, privateKey, nil, db, sugaredLogger, "")
```
Comment (ElliotFriedman) [inline cmd/ebrelayer/main.go:222]: If these parameters aren't needed, let's remove the params from `NewEthereumSub`.

File: cmd/ebrelayer/txs/relayToCosmos.go [hunk 7]
```diff
@@ -49,20 +43,18 @@ func RelayToCosmos(factory tx.Factory, moniker, password string, claims []*types
...
 	}
 
-	sugaredLogger.Infow(
-		"relay sequenceNumber from builder.",
-		"nextSequenceNumber", nextSequenceNumber,
+	sugaredLogger.Infow("RelayToCosmos building, signing, and broadcasting", "messages", messages)
+	// TODO this WithGas isn't correct
+	// TODO we need to investigate retries
+	// TODO we need to investigate what happens when the transaction has already been completed
+	err := tx.BroadcastTx(
```
Comment (ElliotFriedman) [inline cmd/ebrelayer/txs/relayToCosmos.go:50]: Created a card for this in zenhub: https://app.zenhub.com/workspaces/sifchain-603ed608675da3001744d736/issues/sifchain/sifnode/1346 @banshee

File: cmd/ebrelayer/txs/relayToEthereum.go [hunk 8]
```diff
@@ -45,30 +44,32 @@ func RelayProphecyClaimToEthereum(provider string, contractAddress common.Addres
...
-	if err != nil {
-		sugaredLogger.Errorw("failed convert Amount", errorMessageKey, err.Error())
-		return err
+	sugaredLogger.Infow(
+		"Sending new ProphecyClaim to CosmosBridge.",
+		"claim", claim,
+	)
+	// TODO this is leftover from when we were converting from a string; don't need it any more
+	var claimType ethbridge.ClaimType
+	switch claim.ClaimType {
```
Comment (ElliotFriedman) [inline cmd/ebrelayer/txs/relayToEthereum.go:53]: Like this todo says, we can probably change this away from a switch statement to a direct assignment like so:

`claimType = claim.ClaimType`

File: app/app.go (hunk 2 above)
Comment (banshee) [reply app/app.go]: This was removed.

File: cmd/ebrelayer/main.go (hunk 4 above)
Comment (banshee) [reply cmd/ebrelayer/main.go]: Done

File: cmd/ebrelayer/main.go (hunk 5 above)
Comment (banshee) [reply cmd/ebrelayer/main.go:196]: I added more comments

File: cmd/ebrelayer/main.go (hunk 6 above)
Comment (banshee) [reply cmd/ebrelayer/main.go]: Added a comment; this needs to be addressed when we do replay

File: cmd/ebrelayer/txs/relayToEthereum.go (hunk 8 above)
Comment (banshee) [reply cmd/ebrelayer/txs/relayToEthereum.go:46]: Cleaned up this code and added a failure if the claim type isn't lock or burn

**Pull request merged**